docker-app/qfieldcloud/core/views/teams_views.py

```diff
@@ -0,0 +1,265 @@
+    """
+
+    permission_classes = [permissions.IsAuthenticated, TeamPermission]
+    serializer_class = TeamSerializer
+
+    def get_object(self):
+        organization_name = self.kwargs.get("organization_name")
+        team_name = self.kwargs.get("team_name")
+
+        organization = get_object_or_404(Organization, username=organization_name)
+        full_team_name = Team.format_team_name(organization_name, team_name)
+
+        return get_object_or_404(
+            Team, team_organization=organization, username=full_team_name
+        )
+
+    def perform_update(self, serializer):
+        """
+        Handle team name updates with validation.
+        """
+        organization_name = self.kwargs.get("organization_name")
+        new_team_name = self.request.data.get("username")
+
+        if not new_team_name:
+            raise permissions.ValidationError({"error": "New team name is required."})
+
+        full_team_name = Team.format_team_name(organization_name, new_team_name)
+
+        if Team.objects.filter(
+            team_organization=serializer.instance.team_organization,
+            username=full_team_name,
+        ).exists():
+            raise permissions.ValidationError(
+                {"error": "A team with this name already exists."}
+            )
+
+        serializer.save(username=full_team_name)
+
+
+@extend_schema_view(
+    get=extend_schema(description="List all teams under an organization"),
+    post=extend_schema(description="Create a new team under an organization"),
+)
+class TeamListCreateView(generics.ListCreateAPIView):
+    """
+    Handles GET (list all teams) and POST (create a new team) under an organization.
+    """
+
+    permission_classes = [permissions.IsAuthenticated, TeamPermission]
+    serializer_class = TeamSerializer
+
+    def get_queryset(self):
+        organization_name = self.kwargs.get("organization_name")
+        organization = get_object_or_404(Organization, username=organization_name)
+        return Team.objects.filter(team_organization=organization)
+
+    def perform_create(self, serializer):
+        organization_name = self.kwargs.get("organization_name")
+        organization = get_object_or_404(Organization, username=organization_name)
+
+        team_name = self.request.data.get("username")
+
+        if not team_name:
+            raise ValueError({"Empty Value Error": "Team name is required."})
+
+        full_team_name = Team.format_team_name(organization_name, team_name)
+
+        serializer.save(team_organization=organization, username=full_team_name)
+
+
+@extend_schema_view(
+    get=extend_schema(description="Retrieve, update, or delete a team member"),
+    patch=extend_schema(description="Partially update a team member"),
+    put=extend_schema(description="Update a team member"),
+    delete=extend_schema(description="Delete a team member"),
+)
+class GetUpdateDestroyTeamMemberView(generics.RetrieveUpdateDestroyAPIView):
```

Comment:
```suggestion
class GetUpdateDestroyTeamMemberView(generics.RetrieveDestroyAPIView):
```

No need for patching the team member.

Reply:
extend_schema_view also updated